src/libraries/System.Threading/tests/InterlockedTests.cs

```diff
@@ -240,6 +272,54 @@ public void InterlockedExchange_BoxedObject()
             Assert.Equal(12345, (int)value);
         }
 
+        [Fact]
+        public void InterlockedCompareExchange_Int8()
+        {
+            sbyte value = 42;
+
+            Assert.Equal(42, Interlocked.CompareExchange(ref value, 123, 41));
+            Assert.Equal(42, value);
+
+            Assert.Equal(42, Interlocked.CompareExchange(ref value, 123, 42));
+            Assert.Equal(123, value);
+        }
+
+        [Fact]
+        public void InterlockedCompareExchange_UInt8()
+        {
+            byte value = 42;
+
+            Assert.Equal(42u, Interlocked.CompareExchange(ref value, 123, 41));
+            Assert.Equal(42u, value);
+
+            Assert.Equal(42u, Interlocked.CompareExchange(ref value, 123, 42));
+            Assert.Equal(123u, value);
+        }
+
+        [Fact]
+        public void InterlockedCompareExchange_Int16()
+        {
+            short value = 42;
+
+            Assert.Equal(42, Interlocked.CompareExchange(ref value, 12345, 41));
+            Assert.Equal(42, value);
+
+            Assert.Equal(42, Interlocked.CompareExchange(ref value, 12345, 42));
+            Assert.Equal(12345, value);
+        }
+
+        [Fact]
+        public void InterlockedCompareExchange_UInt16()
+        {
+            ushort value = 42;
+
+            Assert.Equal(42u, Interlocked.CompareExchange(ref value, 12345, 41));
+            Assert.Equal(42u, value);
+
+            Assert.Equal(42u, Interlocked.CompareExchange(ref value, 12345, 42));
+            Assert.Equal(12345u, value);
+        }
+
```

Comment:
I think it'd be good to add some tests that use our BoundedMemory helper to ensure we're not touching memory outside of the single value being exchanged. Some of the proposed implementations in the past would have erroneously done so.

Reply:
I'll take a look tomorrow, opened the PR after running the tests on linux arm64 to see how other platforms react with the CI for now.

Reply:
Done.